addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -114,52 +110,72 @@ private synchronized void login() {
-                wr.write(postData);
+            if (httpClient.isStarted()) {
+                httpClient.stop();
             }
+            httpClient.start();
 
-            // get cookie
-            String headerName;
-            for (int i = 1; (headerName = connection.getHeaderFieldKey(i)) != null; i++) {
-                if (headerName.equals("Set-Cookie")) {
-                    cookie = connection.getHeaderField(i);
-                    break;
-                }
-            }
+            url = TAHOMA_URL + "login";
+            String urlParameters = "userId=" + thingConfig.getEmail() + "&userPassword=" + thingConfig.getPassword();
 
-            InputStream response = connection.getInputStream();
-            String line = readResponse(response);
+            ContentResponse response = httpClient.newRequest(url)
+                    .method(HttpMethod.POST)
+                    .header(HttpHeader.ACCEPT_LANGUAGE, "en-US,en")
+                    .header(HttpHeader.ACCEPT_ENCODING, "gzip, deflate")
+                    .header("X-Requested-With", "XMLHttpRequest")
+                    .agent(TAHOMA_AGENT)
+                    .content(new StringContentProvider(urlParameters), "application/x-www-form-urlencoded; charset=UTF-8")
+                    .send();
 
-            SomfyTahomaLoginResponse data = gson.fromJson(line, SomfyTahomaLoginResponse.class);
+            SomfyTahomaLoginResponse data = gson.fromJson(response.getContentAsString(), SomfyTahomaLoginResponse.class);
 
             if (data.isSuccess()) {
-                String version = data.getVersion();
-                logger.debug("SomfyTahoma cookie: {}", cookie);
-                logger.debug("SomfyTahoma version: {}", version);
+                logger.debug("SomfyTahoma version: {}", data.getVersion());
                 updateStatus(ThingStatus.ONLINE);
             } else {
-                logger.debug("Login response: {}", line);
+                logger.debug("Login response: {}", response.getContentAsString());
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Error logging in");
-                throw new SomfyTahomaException(line);
+                throw new SomfyTahomaException(response.getContentAsString());
             }
-        } catch (MalformedURLException e) {
-            logger.error("The URL '{}' is malformed!", url, e);
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                    "The URL '" + url + "' is malformed: ");
-        } catch (Exception e) {
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
             logger.error("Cannot get login cookie!", e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Cannot get login cookie");
+        } catch (Exception e) {
+            logger.error("Cannot start http client", e);
```

Comment:
The user already see the ThingStatusDetail with this message, so no need to log it into the logfile. Please change it to debug or remove it.

Reply:
OK